core/src/main/java/jenkins/model/logging/LogBrowser.java

```diff
@@ -0,0 +1,105 @@
+package jenkins.model.logging;
+
+import hudson.console.AnnotatedLargeText;
+import hudson.console.ConsoleNote;
+import jenkins.util.io.OnMaster;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.Beta;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Reader;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Defines how the logs should be browsed.
+ * @author Oleg Nenashev
+ * @author Jesse Glick
+ * @since TODO
+ */
+@Restricted(Beta.class)
+public abstract class LogBrowser<T extends Loggable> extends LogHandler implements OnMaster {
+
+    public LogBrowser(Loggable loggable) {
+        super(loggable);
+    }
+
+    /**
+     * Gets log for an object.
+     * @return Created log or {@link jenkins.model.logging.impl.BrokenAnnotatedLargeText} if it cannot be retrieved
+     */
+    @Nonnull
+    public abstract AnnotatedLargeText<T> overallLog();
+
+    /**
+     * Gets log for a part of the object.
+     * @param stepId Identifier of the step to be displayed.
+     *               It may be Pipeline step or other similar abstraction
```

Comment:
Why does this need to be defined here? Seems like it belongs only in `external-logging-api`.

Reply:
"Steps" may exist for custom logging implementations , eventually not only for Pipeline. So I would prefer to keep API here at lower level